evap/student/templates/student_vote.html

```diff
@@ -240,29 +249,34 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
                 return false;
             });
 
-            function markNoAnswerFor(contributorId) {
-                // uncheck all other options and deactivate buttons
-                $("#vote-area-" + contributorId + " div.vote-inputs [type=radio][value!=6]").removeAttr("checked");
-                $("#vote-area-" + contributorId + " div.vote-inputs [type=radio][value!=6]").parent().removeClass("active");
+            document.querySelectorAll("[data-mark-no-answers-for]").forEach(button => {
+                button.addEventListener("click", () => {
+                    const contributorId = button.dataset.markNoAnswersFor;
+                    const voteArea = document.querySelector(`#vote-area-${contributorId}`);
+                    // check "no answer" and uncheck all other options
+                    voteArea.querySelectorAll(".vote-inputs [type=radio][value='6']")
+                        .forEach(radioInput => radioInput.click());
 
-                // check "no answer" and activate the button
-                $("#vote-area-" + contributorId + " div.vote-inputs [type=radio][value=6]").prop('checked', true);
-                $("#vote-area-" + contributorId + " div.vote-inputs [type=radio][value=6]").parent().addClass("active");
+                    // hide questionnaire for contributor
+                    $(voteArea).collapse("hide");
 
-                // hide questionnaire for contributor
-                $("#vote-area-" + contributorId).collapse("hide");
+                    // setting the 'checked' prop does not trigger a change event, so manually save all data here
+                    sisyphus.saveAllData();
```

Comment:
you are not setting the `checked` property manually anymore. is the manual save still required?

Reply:
Nice catch, sisyphus saves implicitly :)